File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int count(int n){
    int res = 0;
    for(int i = 1; i < int(sqrt(n))+1; i++){
        if(n%i==0) res += 2;
    }
    if (n == int(sqrt(n))*int(sqrt(n))) res-=1;
    return res;
}

int solution(int left, int right) {
    int answer = 0;
    for(int i = left; i <= right; i++){
        if(count(i)%2) answer -= i;
        else answer += i;
    }
    return answer;
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def cnt(n):
    res = 0
    for i in range(1,n+1):
        if n%i == 0:
            res += 1
    return res

def solution(left, right):
    answer = 0
    for i in range(left, right+1):
        if cnt(i)%2==0:
            answer += i
        else:
            answer -= i
    return answer